test/test_mypy.py

```diff
@@ -324,6 +336,33 @@ def test_typeddict_document_type_insertion(self) -> None:
         )
         coll.insert_many([bad_movie])
 
+    @only_type_check
+    def test_typeddict_explicit_document_type(self) -> None:
+        out = MovieWithId(_id=ObjectId(), name="THX-1138", year=1971)
```

Comment:
Shouldn't these be find operation results?

Reply:
Okay, I think instead we should have a single test that uses `NotRequired` and actually does execute a `find`

Reply:
I don't think we need these tests since they aren't PyMongo-specific

Reply:
Fair enough